### PR TITLE
Add ability to augment classes with fields from other classes in Painless

### PR DESCRIPTION
**Pull request by jdconrad**
This change adds a a new "augmented" annotation to the Painless allowlist parser. The first use of the annotation supports adding `static final` fields to a specified allowlist class from another class. This supports the fields api as we can add additional fields types from other classes and augment the Field class with the new types.

### CODE REVIEW

File: modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/annotation/AugmentedAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.painless.spi.annotation;
+
```
Comment (stu-elastic) [inline modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/annotation/AugmentedAnnotation.java:10]: Javadoc?

**Review by stu-elastic — APPROVED** [review]
```
class org.elasticsearch.script.Field {
    org.elasticsearch.script.Converter UnsignedLong @augmented[augmented_canonical_class_name="org.elasticsearch.xpack.unsignedlong.UnsignedLongField"]
}

class org.elasticsearch.xpack.unsignedlong.UnsignedLongField @no_import {
}
```

Works with this change, the bottom class is necessary to avoid a `class_not_found_exception`

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review! Will add a Javadoc as requested.

File: modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/annotation/AugmentedAnnotation.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/annotation/AugmentedAnnotation.java:10]: Added.

**Pull request merged**